test/e2e/performanceprofile/functests/2_performance_update/tuned_deferred.go

```diff
@@ -0,0 +1,320 @@
+				"tuned.openshift.io/deferred": deferMode,
+			},
+		},
+		Spec: tunedv1.TunedSpec{
+			Profile: []tunedv1.TunedProfile{
+				{
+					Name: &tunedName,
+					Data: &data,
+				},
+			},
+			Recommend: []tunedv1.TunedRecommend{
+				{
+					MachineConfigLabels: map[string]string{"machineconfiguration.openshift.io/role": testutils.RoleWorkerCNF},
+					Priority:            &priority,
+					Profile:             &tunedName,
+				},
+			},
+		},
+	}
+
+	testlog.Infof("Creating Tuned Profile Patch: %s", name)
+	Expect(testclient.ControlPlaneClient.Create(context.TODO(), tuned)).To(Succeed())
+	return tuned, nil
+}
+
+func execSysctlOnWorkers(ctx context.Context, workerNodes []corev1.Node, sysctlMap map[string]string) {
+	var err error
+	var out []byte
+	isSNO := false
+	for _, node := range workerNodes {
+		for param, expected := range sysctlMap {
+			By(fmt.Sprintf("executing the command \"sysctl -n %s\"", param))
+			tunedCmd := []string{"/bin/sh", "-c", "chroot /host sysctl -a | grep shmmni"}
+			tunedPod := nodes.TunedForNode(&node, isSNO)
+			out, err = pods.WaitForPodOutput(context.TODO(), testclient.K8sClient, tunedPod, tunedCmd)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(strings.TrimSpace(strings.SplitN(string(out), "=", 2)[1])).Should(Equal(expected), "parameter %s value is not %s.", out, expected)
+		}
+	}
+}
+
+func verifyImmediateApplication(tuned *tunedv1.Tuned, workerCNFNodes []corev1.Node, kernel_shmmni string) {
+	time.Sleep(40 * time.Second)
+	sysctlMap := map[string]string{
+		"kernel.shmmni": kernel_shmmni,
+	}
+	execSysctlOnWorkers(context.TODO(), workerCNFNodes, sysctlMap)
+}
+
+func verifyDeferredApplication(tuned *tunedv1.Tuned, workerCNFNodes []corev1.Node, kernel_shmmni string, name string) (bool, error) {
+	time.Sleep(20 * time.Second)
+	expectedMessage := fmt.Sprintf("The TuneD daemon profile is waiting for the next node restart: %s", name)
+	profile := &tunedv1.Profile{}
+	node := workerCNFNodes[0]
+	err := testclient.DataPlaneClient.Get(context.TODO(), client.ObjectKey{Name: node.Name, Namespace: tuned.Namespace}, profile)
+	if err != nil {
+		return false, fmt.Errorf("failed to get profile: %w", err)
+	}
+
+	for _, condition := range profile.Status.Conditions {
+		if condition.Message == expectedMessage {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func rebootNode(poolName string) {
+	profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+	if err != nil {
+		testlog.Errorf("Unable to fetch latest performance profile err: %v", err)
+	}
+	testlog.Info("Rebooting the node")
+	// reboot the node, for that we change the numa policy to best-effort
+	// Note: this is used only to trigger reboot
+	policy := "best-effort"
+	// Need to make some changes to pp , causing system reboot
+	currentPolicy := profile.Spec.NUMA.TopologyPolicy
+	if *currentPolicy == "best-effort" {
+		policy = "single-numa-node"
+	}
+	profile.Spec.NUMA = &performancev2.NUMA{
+		TopologyPolicy: &policy,
+	}
```

Comment:
this is suspicious though, and this doesn't just reboot the node, but actually changes the config - which is also never reverted?

Reply:
Updated with changes to restore the original configuration upon test completion. A mechanism for rebooting is required anyhow, since this change does not directly impact the associated test, I went with this way.